site/src/pages/ManagementSettingsPage/IdpSyncPage/IdpSyncPage.tsx

```diff
@@ -37,6 +41,36 @@ export const IdpSyncPage: FC = () => {
 			],
 		});
 
+	useEffect(() => {
+		if (groupIdpSyncSettingsQuery.error) {
+			displayError(
+				getErrorMessage(
+					groupIdpSyncSettingsQuery.error,
+					"Unable to load group IdP sync settings.",
+				),
+			);
+		}
+	}, [groupIdpSyncSettingsQuery.error]);
+
+	useEffect(() => {
+		if (roleIdpSyncSettingsQuery.error) {
+			displayError(
+				getErrorMessage(
+					roleIdpSyncSettingsQuery.error,
+					"Unable to load role IdP sync settings.",
+				),
+			);
+		}
+	}, [roleIdpSyncSettingsQuery.error]);
+
+	useEffect(() => {
+		if (groupsQuery.error) {
+			displayError(
+				getErrorMessage(groupsQuery.error, "Unable to load groups."),
+			);
+		}
+	}, [groupsQuery.error]);
```

Comment:
Do we want these set up as effects? I think there's a risk that these will run too often

I just tested this out, and if React Query ever refetches and errors out, it will update the memory reference of the `error` properties on these queries. Even if the query already had an error, and even if the error contents are exactly the same, a re-render will happen, and the effects will re-trigger.  So the user could have toast messages reappear right after they've just dismissed them. I could see that being frustrating

Similarly, what do we want to happen when multiple of these error out? Our toast message setup only shows the most recent message

Reply:
My preference would be to bring these into the route-level UI. The RQ team also strongly cautions against `useEffect` state syncs like these (which is why they removed the hook-level `onError` handler)